lib/cartopy/io/ogc_clients.py

```diff
@@ -232,8 +232,8 @@ def _image_and_extent(self, wms_proj, wms_srs, wms_extent, output_proj,
         return LocatedImage(wms_image, extent)
 
     def fetch_raster(self, projection, extent, target_resolution):
-        service = self.service
         min_x, max_x, min_y, max_y = extent
+        target_resolution = [np.ceil(val) for val in target_resolution]
```

Comment:
I am not familiar with this either, but if taking the ceiling means getting more than request that is better than being cropped. Right?